agent/src/main.cpp

```diff
@@ -155,7 +145,7 @@ int main(int argc, char **argv)
     dbusDispatcher.attach(mainContext->gobj());
 
     try {
-        DBus::Connection connection = getBusConnection(useSystemBus);
+        DBus::Connection connection = getBusConnection(!useSessionBus);
```

Comment:
it may be better if we use this as `getBusConnection(useSessionBus);`
and change  `DBus::Connection getBusConnection(bool useSystemBus)` to get  an argument as `bool useSessionBus`